zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -555,7 +566,11 @@ angular.module('zeppelinWebApp')
     if (isNaN(timeMs) || timeMs < 0) {
       return '&nbsp;';
     }
-    return 'Took ' + (timeMs/1000) + ' seconds';
+    var desc = 'Took ' + (timeMs/1000) + ' seconds.';
+    if (pdata.textSaved !==undefined && Date.parse(pdata.textSaved) > Date.parse(pdata.dateStarted)){
+      desc += ' (outdated)'
+    }
+    return desc;
```

Comment:
nit: just a thought, would there be a concern since the save button is only showing when needed, that the other button on the right (hide result, etc) would be shifting as it goes?
IMO it would be great if they don't move around for usability because of muscle memory etc.